cluster-autoscaler/utils/nodegroupset/scale_up.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroupset
+
+import (
+	"fmt"
+	"sort"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+
+	"github.com/golang/glog"
+)
+
+// ScaleUpInfo contains information about planned scale-up of a single NodeGroup
+type ScaleUpInfo struct {
```

Comment:
I'm returning it from BalanceScaleUpBetweenGroups, which is a public method (in fact I was planning to call it from core/scale_up.go).